exchanges/okcoin/okcoin_websocket.go

```diff
@@ -147,150 +238,268 @@ func (o *OKCoin) WsHandleData(respRaw []byte) error {
 	var eventResponse WebsocketEventResponse
 	err = json.Unmarshal(respRaw, &eventResponse)
 	if err == nil && eventResponse.Event != "" {
-		if eventResponse.Event == "login" {
-			if o.Websocket.Match.Incoming("login") {
-				o.Websocket.SetCanUseAuthenticatedEndpoints(eventResponse.Success)
+		switch eventResponse.Event {
+		case "login":
+			if o.Websocket.Match.IncomingWithData("login", respRaw) {
+				o.Websocket.SetCanUseAuthenticatedEndpoints(eventResponse.Code == "0")
+			}
+		case "subscribe", "unsubscribe":
+			o.Websocket.DataHandler <- eventResponse
+		case "error":
+			waitingSignatureLock.Lock()
+			for x := range waitingSignatures {
+				if strings.Contains(dataResponse.Message, waitingSignatures[x]) {
+					o.Websocket.Match.IncomingWithData(waitingSignatures[x], respRaw)
+					return nil
+				}
+			}
+			waitingSignatureLock.Unlock()
```

Comment:
Why do we only want to match the response with errors? I'm not seeing `waitingSignatures` match anywhere else.

Reply:
Yes, I had to remove the signature from the list above when response data with a non-empty ID is found like below.
```
if dataResponse.ID != "" {
		waitingSignatureLock.Lock()
		delete(waitingSignatures, dataResponse.ID)
		waitingSignatureLock.Unlock()
		if !o.Websocket.Match.IncomingWithData(dataResponse.ID, respRaw) {
			return fmt.Errorf("couldn't match incoming message with id: %s and operation: %s", dataResponse.ID, dataResponse.Operation)
		}
		return nil
}
```
👍🏽.